src/docs/arch-tiered-storage.dox

```diff
@@ -0,0 +1,439 @@
+- \c REMOVE_LOCAL: remove a local file that is has already been copied
+    to the cloud
+- \c REMOVE_SHARED: remove a cloud file, perhaps as part of a drop operation
+
+There are individual functions to add items of each type, and to get
+the next item, if any, or each type. That allows us to process items
+in a certain order, if we wish. With only one server thread, all operations
+occur sequentially. It would make sense in the future
+to do \c FLUSH operations in parallel.
+
+@section ts_metadata Metadata and Associated Data Structures
+
+@subsection ts_metadata_non Non-tiered Tables
+
+When non-tiered table \c A is created (without named column groups),
+there are two entries in the metadata Btree, having these keys:
+- \c "table:A"
+- \c "file:A.wt"
+
+For brevity, we are showing just the keys; the values in the metadata
+Btree are configuration strings that would show information about
+key/value format and other creation options, checkpoints associated with the table, etc.
+
+The \c "table:A" represents the table interface that the API caller
+uses, and the associated data structure in memory for the table is \c WT_TABLE.  If
+there were column groups or indices, there may be multiple Btrees
+associated with the table, and listed in the metadata,
+but in this example there is just one.  The
+\c "file:" entry represents the Btree stored on disk, as well as parts
+cached in memory.  The data structure implied by this entry is a \c
+WT_BTREE.
+
+In summary, the relationship between prefixes and data structures is as follows:
+
+URI prefix    | struct type    | has dhandle?  |  dhandle notes
+------------- | -------------- | ------------- | --------------------------------------
+\c table:     | WT_TABLE       | yes           | the dhandle is cast to (WT_TABLE *)
+\c file:      | WT_BTREE       | yes           | dhandle->handle is a (WT_BTREE *)
+
+@subsection ts_metadata_tiered Tiered Tables
+
+For a tiered table \c A that has gone through a few flush_tiers, we might have the following
+entries in the metadata table:
+
+- \c "table:A"
+- \c "file:A-0000000004.wtobj"
+- \c "object:A-0000000003.wtobj"
+- \c "object:A-0000000002.wtobj"
+- \c "object:A-0000000001.wtobj"
+- \c "tier:A"
+- \c "tiered:A"
+
+The \c "table:A" entry, like before, is the "top-level" URI that is used for
+API calls.  As before, it maps to a \c WT_TABLE, and the table has the
+\c WT_TABLE->is_tiered_shared flag set.  Given this flag setting,
+WiredTiger knows to construct a matching \c "tiered:A" entry, which relates to
+a \c WT_TIERED struct.
+
+A \c WT_TIERED has most of the useful information about a tiered table.
+In particular the Btree is found in the dhandle. To be precise,
+\c WT_TIERED->iface is the dhandle, and \c WT_TIERED->iface.handle is
+a pointer to a WT_BTREE.  The \c WT_TIERED structure is also ready to
+support union tables, see future work in @ref ts_future_sharing.
+The local tier part of union tables would in WT_TIERED->tiers[WT_TIERED_INDEX_LOCAL],
```

Comment:
@sueloverso You are right, it is "file:A-00000001.wtobj" in the 0'th slot. 
 But honestly, this is a little confusing to me, because the uri "file:A-000000N.wtobj" is weird in the tiered system.  The checkpoint in the metadata for each "file:A-000000N.wtobj" entry is the checkpoint for the original empty file when it was created, and it's never updated.  So if you open it, the file is empty!  Some of this is in https://jira.mongodb.org/browse/WT-12952, I just confirmed and can demo the problem if you are curious.

So I guess having the INDEX_LOCAL slot having "file:A-00000001.wtobj" is convenient to have the filename to push the file to the cloud?  But it shouldn't really be interpreted as a usable entry in the metadata.

I'll try to write some sensible text without getting into the oddness I see.

Reply:
Just rewrote.